packages/cli/src/modules/chat-hub/chat-hub.service.ts

```diff
@@ -375,51 +382,45 @@ export class ChatHubService {
 	}
 
 	async sendHumanMessage(res: Response, user: User, payload: HumanMessagePayload) {
-		const { sessionId, messageId, message, model, credentials, previousMessageId } = payload;
+		const { sessionId, messageId, message, model, credentials, previousMessageId, attachments } =
+			payload;
 		const { provider } = model;
 
+		const workflowId = generateNanoId();
 		const selectedModel = this.getModelWithCredentials(model, credentials);
 
+		// Store attachments early to populate 'id' field via BinaryDataService
+		const processedAttachments = await this.chatHubAttachmentService.store(attachments, workflowId);
```

Comment:
Did you consider storing these late instead, starting the execution as soon as possible with just IBinaryData constructed from the incoming message?

Reply:
The store method [drops](https://github.com/n8n-io/n8n/blob/chat-45-5/packages/core/src/binary-data/binary-data.service.ts#L133) `data` field, so `processedAttachments` doesn't contain file itself anymore.

If we store binary data afterwards, the execution data will contain the base64-encoded file regardless of storage type, which might be fine base models but I think it's not good for n8n workflow agents.